chatkit/src/main/java/com/pusher/chatkit/ChatManager.kt

```diff
@@ -181,25 +195,19 @@ data class Attachment(
 )
 
 data class Cursor(
-    val userId: String,
-    val roomId: Int,
+    override val userId: String,
+    override val roomId: Int,
     val type: Int,
     val position: Int,
     val updatedAt: String
-)
-
-fun UserService.userFor(cursor: Cursor): Promise<Result<User, Error>> =
-    fetchUserBy(cursor.userId)
-
-fun RoomService.roomFor(cursor: Cursor): RoomPromiseResult =
-    fetchRoomBy(cursor.roomId)
+) : HasUser, HasRoom
 
 data class ChatEvent(
     val eventName: String,
-    val userId: String? = null,
+    override val userId: String = "",
```

Comment:
Not all events provide a user ID. In fact I _think_ it's just typing indicator ones. Having it as `String?` perhaps makes more sense? Or maybe making a separate class for `ChatEvent` type things that have user IDs?

Reply:
Actually, this field is never used when parsing. I want to move these to use a better way of parsing the events (sealed classes), I'll leave it for now as a reference